### PR TITLE
Feature/id 366/filter incidents

### PR DESCRIPTION
**Pull request by luiggy-mamani-jalafund**
What I did?
Send incidents using filters by type of incident.
The filters, filter by one or more types and you can use them to filter with the nearby incidents filter that had been created.

How I dis?
Using the find method and putting the filter-like query parameter

Why I did?
To send this data when the user selects a type of incident to see

### CODE REVIEW

**Review by sebastian-barra-jalafund — APPROVED** [review]

File: src/controller/responses/incidentRes.js [hunk 1]
```diff
@@ -14,54 +14,73 @@ export const sendIncidentById = async (request, response) => {
...
  * Retrieves nearby incidents based on the provided longitude, latitude, and radius.
- * 
+ *
  * @param {number} longitude - The longitude coordinate.
  * @param {number} latitude - The latitude coordinate.
  * @param {number} radius - The radius in which to search for incidents.
  * @returns {Array} - The array of nearby incidents.
  */
-const getNearIncidents = async (longitude, latitude, radius) => {
+const getNearIncidents = async (types, longitude, latitude, radius) => {
```
Comment (zrock-dev) [inline src/controller/responses/incidentRes.js:56]: By adding this parameter, you are constraining the queries to type, but what if we want to have all nearby incidents?

File: src/controller/responses/incidentRes.js [hunk 2]
```diff
@@ -14,54 +14,73 @@ export const sendIncidentById = async (request, response) => {
...
+		return types
+			? await Incident.find({
+					type: getTypeQuery(types),
+					'geometry.coordinates': nearIncidentQuery
+			  })
+			: await Incident.find({
+					'geometry.coordinates': nearIncidentQuery
+			  });
 	} catch (err) {
 		return [];
```
Comment (zrock-dev) [inline src/controller/responses/incidentRes.js]: Could you log this error? So there will be a registry.

**Review by zrock-dev — CHANGES_REQUESTED** [review]
The name of the file `src/controller/responses/incidentRes.js` is pretty ambiguous since it can be interpreted as `incidentResources`, `incidentResponses` or even `incidentResults`

File: src/controller/responses/incidentRes.js (hunk 1 above)
Comment (luiggy-mamani-jalafund) [reply src/controller/responses/incidentRes.js:56]: the method has a conditional if the types query is null, in that case the method just return the near incidents

Comment (zrock-dev) [reply src/controller/responses/incidentRes.js:56]: cool, it's solved then

**Review by zrock-dev — APPROVED** [review]

**Review by saletG08 — APPROVED** [review]
I don't have any comments

**Pull request merged**